diff --git a/profiles/ding2/modules/bpi/bpi.admin.inc b/profiles/ding2/modules/bpi/bpi.admin.inc
index d45c7b88..ef199498 100644
--- a/profiles/ding2/modules/bpi/bpi.admin.inc
+++ b/profiles/ding2/modules/bpi/bpi.admin.inc
@@ -179,6 +179,7 @@ function bpi_admin_content_mapping_form($form, $form_state) {
     '#title' => t('BPI teaser'),
     '#description' => t('The field to extract the teaser from. If the content type have a body summary, assign it to the body field.'),
     '#options' => $field_instances,
+    '#empty_value' => '',
     '#default_value' => variable_get('bpi_field_teaser', ''),
   );
 
@@ -187,6 +188,7 @@ function bpi_admin_content_mapping_form($form, $form_state) {
     '#title' => t('BPI body'),
     '#description' => t('Field to extract the main cotenten from (body field).'),
     '#options' => $field_instances,
+    '#empty_value' => '',
     '#default_value' => variable_get('bpi_field_body', ''),
   );
 
@@ -195,6 +197,7 @@ function bpi_admin_content_mapping_form($form, $form_state) {
     '#title' => t('BPI tags'),
     '#description' => t('Field used to get tags from.'),
     '#options' => $field_instances,
+    '#empty_value' => '',
     '#default_value' => variable_get('bpi_field_tags', ''),
   );
 
@@ -203,6 +206,7 @@ function bpi_admin_content_mapping_form($form, $form_state) {
     '#title' => t('BPI materials'),
     '#description' => t('Field used to get reference to the T!NG data well.'),
     '#options' => $field_instances,
+    '#empty_value' => '',
     '#default_value' => variable_get('bpi_field_materials', ''),
   );
 
