diff --git a/ding2.make b/ding2.make
index 59cb99e1..d05bea21 100644
--- a/ding2.make
+++ b/ding2.make
@@ -353,7 +353,7 @@ projects[ask_vopros][download][tag] = "1.1"
 libraries[bpi-client][destination] = "modules/bpi/lib"
 libraries[bpi-client][download][type] = "git"
 libraries[bpi-client][download][url] = "http://github.com/ding2/bpi-client.git"
-libraries[bpi-client][download][branch] = "master"
+libraries[bpi-client][download][branch] = "7.x-2.4.0-rc3"
 
 libraries[ckeditor][download][type] = "get"
 libraries[ckeditor][download][url] = http://download.cksource.com/CKEditor/CKEditor/CKEditor%204.4.7/ckeditor_4.4.7_full.zip
