diff --git a/profiles/ding2/modules/ding_user_form/ding_user_form.module b/profiles/ding2/modules/ding_user_form/ding_user_form.module
index fdb57079..da4027c0 100644
--- a/profiles/ding2/modules/ding_user_form/ding_user_form.module
+++ b/profiles/ding2/modules/ding_user_form/ding_user_form.module
@@ -173,6 +173,11 @@ function ding_user_form_user_login_validate($form, &$form_state) {
       $form_state['uid'] = $account->uid;
       // We're not calling user_login_submit like user_external_login_register
       // does, it's already the submit handler.
+
+      // PATCHED: to load retailer information.
+      if (ding_provider_implements('user', 'authenticate_finalize')) {
+        ding_provider_invoke('user', 'authenticate_finalize', $account);
+      }
     }
   }
   catch (DingProviderUserAuthFailure $e) {
