diff --git a/profiles/ding2/modules/ding_base/ding_base.info b/profiles/ding2/modules/ding_base/ding_base.info
index d6d039a5..4bc3c702 100644
--- a/profiles/ding2/modules/ding_base/ding_base.info
+++ b/profiles/ding2/modules/ding_base/ding_base.info
@@ -19,11 +19,6 @@ configure = admin/config/ding
 features[ctools][] = page_manager:pages_default:1
 features[ctools][] = strongarm:strongarm:1
 features[features_api][] = api:2
-features[variable][] = bpi_content_type
-features[variable][] = bpi_field_body
-features[variable][] = bpi_field_materials
-features[variable][] = bpi_field_teaser
-features[variable][] = bpi_field_title
 features[variable][] = date_api_use_iso8601
 features[variable][] = date_api_version
 features[variable][] = date_db_tz_support
@@ -130,4 +125,3 @@ features[variable][] = varnish_bantype
 features[variable][] = varnish_cache_clear
 features[variable][] = varnish_flush_cron
 features[variable][] = varnish_version
-mtime = 1486469665
diff --git a/profiles/ding2/modules/ding_base/ding_base.strongarm.inc b/profiles/ding2/modules/ding_base/ding_base.strongarm.inc
index 01a4c861..7fc6af80 100644
--- a/profiles/ding2/modules/ding_base/ding_base.strongarm.inc
+++ b/profiles/ding2/modules/ding_base/ding_base.strongarm.inc
@@ -10,41 +10,6 @@
 function ding_base_strongarm() {
   $export = array();
 
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'bpi_content_type';
-  $strongarm->value = 'ding_news';
-  $export['bpi_content_type'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'bpi_field_body';
-  $strongarm->value = 'field_ding_news_body';
-  $export['bpi_field_body'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'bpi_field_materials';
-  $strongarm->value = 'field_ding_news_materials';
-  $export['bpi_field_materials'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'bpi_field_teaser';
-  $strongarm->value = 'field_ding_news_lead';
-  $export['bpi_field_teaser'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'bpi_field_title';
-  $strongarm->value = 'title';
-  $export['bpi_field_title'] = $strongarm;
-
   $strongarm = new stdClass();
   $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
   $strongarm->api_version = 1;
@@ -69,7 +34,6 @@ function ding_base_strongarm() {
   $strongarm = new stdClass();
   $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
   $strongarm->api_version = 1;
-
   $strongarm->name = 'ding_varnish_content_types';
   $strongarm->value = array(
     'ding_event' => 'ding_event',
