diff --git a/modules/ding_loan/plugins/content_types/loans.inc b/modules/ding_loan/plugins/content_types/loans.inc
index b3eff4c..19664b2 100644
--- a/modules/ding_loan/plugins/content_types/loans.inc
+++ b/modules/ding_loan/plugins/content_types/loans.inc
@@ -129,6 +129,7 @@ function ding_loan_loans_form($form, &$form_state, $account, $loans = array()) {
 
       $item = array(
         '#type' => 'material_item',
+        '#loan' => $loan,
         '#id' => $loan->id,
         '#title' => $title,
         '#cover' => field_view_field('ting_object', $entity, 'ting_cover', 'user_list'),
