diff --git a/profiles/ding2/modules/ding_user/ding_user.module b/profiles/ding2/modules/ding_user/ding_user.module
index 4f88c3bd..436d6b6d 100644
--- a/profiles/ding2/modules/ding_user/ding_user.module
+++ b/profiles/ding2/modules/ding_user/ding_user.module
@@ -872,7 +872,10 @@ function ajax_command_ding_user_authenticate($extra_data, $redirect_uri = '') {
     }

     foreach ($modules as $module) {
-      $command = module_invoke($module, $hook, $redirect_uri);
+      $cmd = module_invoke($module, $hook, $redirect_uri);
+      if (is_array($cmd)) {
+        $command = $cmd;
+      }
     }
   }
   else {
diff --git a/profiles/ding2/modules/ding_user/ding_user.module b/profiles/ding2/modules/ding_user/ding_user.module
index 436d6b6d..4e8cea46 100644
--- a/profiles/ding2/modules/ding_user/ding_user.module
+++ b/profiles/ding2/modules/ding_user/ding_user.module
@@ -878,7 +878,8 @@ function ajax_command_ding_user_authenticate($extra_data, $redirect_uri = '') {
       }
     }
   }
-  else {
+
+  if (!isset($command)) {
     // Change default ajax action to default login form's if https is enabled.
     if (variable_get('https', FALSE)) {
       $form = drupal_get_form('user_login');
diff --git a/profiles/ding2/modules/ding_user/ding_user.module b/profiles/ding2/modules/ding_user/ding_user.module
index 4e8cea46..c06898de 100644
--- a/profiles/ding2/modules/ding_user/ding_user.module
+++ b/profiles/ding2/modules/ding_user/ding_user.module
@@ -872,6 +872,16 @@ function ajax_command_ding_user_authenticate($extra_data, $redirect_uri = '') {
     }

     foreach ($modules as $module) {
+      // This is patch for eReolen Go that uses that fact the last called hook
+      // can override the other hooks as the commend only stores the last
+      // returned array from the hooks. This is used to ensure that (on go)
+      // adgangsplatformen don't hijack the login ajax popup with the two SSO
+      // login options. In DDB CMS and on eReolen there is only on SSO and here
+      // we what the default behaviour. But when ding_adgangsplatformen is
+      // disabled on eRolen go the hook in breol_base will return null and set
+      // the $command to NULL. Which then will make an JS error stopping th
+      // efault login popup. So this check and the check in line #890 below
+      // ensure default non SSO behaviour.
       $cmd = module_invoke($module, $hook, $redirect_uri);
       if (is_array($cmd)) {
         $command = $cmd;
