diff --git a/profiles/ding2/modules/ding_loan/plugins/content_types/loans.inc b/profiles/ding2/modules/ding_loan/plugins/content_types/loans.inc
index 1b63087c..115cb14b 100644
--- a/profiles/ding2/modules/ding_loan/plugins/content_types/loans.inc
+++ b/profiles/ding2/modules/ding_loan/plugins/content_types/loans.inc
@@ -32,7 +32,8 @@ function ding_loan_loans_content_type_render($subtype, $conf, $panel_args, $cont
   $account = isset($context->data) ? $context->data : NULL;
 
   $loans = FALSE;
-  $cache_key = 'list-' . $conf['overdue'] . '-' . $conf['regular'];
+  // Don't pollute the log with "Undefined index" warnings.
+  $cache_key = isset($conf['overdue'], $conf['regular']) ? 'list-' . $conf['overdue'] . '-' . $conf['regular'] : 'list-' . '' . '-' . '';
   if (module_exists('ding_session_cache')) {
     $loans = ding_session_cache_get('ding_loan', $cache_key);
   }
