diff --git a/modules/ding_user/ding_user.module b/modules/ding_user/ding_user.module
index 147f9ad9c..0bb4e7f6f 100644
--- a/modules/ding_user/ding_user.module
+++ b/modules/ding_user/ding_user.module
@@ -123,10 +123,13 @@ function ding_user_authenticate(array $user_info) {
   $account = FALSE;
 
   try {
+    // Check if more info is provide from the user login form or SSO provider and default to empty array.
+    $extra = $user_info['extra'] ?? [];
+
     if (array_key_exists('single_sign_on', $user_info) && $user_info['single_sign_on'] == TRUE) {
       // No password provided by the authentication request (not even the empty
       // string), so we assume that this is a single-sign-on request.
-      $auth_res = ding_provider_invoke('auth', 'single_sign_on', $user_info['name']);
+      $auth_res = ding_provider_invoke('auth', 'single_sign_on', $user_info['name'], $extra);
 
       // Store value in session to be used later on to check if user was signed
       // in with SSO.
@@ -134,7 +137,7 @@ function ding_user_authenticate(array $user_info) {
     }
     else {
       // Normal user log in request.
-      $auth_res = ding_provider_invoke('user', 'authenticate', $user_info['name'], $user_info['pass']);
+      $auth_res = ding_provider_invoke('user', 'authenticate', $user_info['name'], $user_info['pass'], $extra);
     }
   }

