diff --git b/modules/ting_search/plugins/content_types/search_result.inc a/modules/ting_search/plugins/content_types/search_result.inc
index ed723e2..21ed17f 100644
--- b/modules/ting_search/plugins/content_types/search_result.inc
+++ a/modules/ting_search/plugins/content_types/search_result.inc
@@ -89,10 +89,10 @@ function ting_search_search_result_content_type_render($subtype, $conf, $panel_a
       'rank_creator' => t('Creator'),
       'rank_subject' => t('Subject')
     );
-    if (isset($search_result) && in_array($search_result->sortUsed, array_keys($frequency_rank))) {
-      $msg = t('Records are sorted by %sortUsed . Select another sort if it is not suited', array('%sortUsed' => $frequency_rank[$search_result->sortUsed]));
-      $message .= '<div class="messages status">' . $msg . '</div>';
-    }
+    // if (isset($search_result) && in_array($search_result->sortUsed, array_keys($frequency_rank))) {
+    //   $msg = t('Records are sorted by %sortUsed . Select another sort if it is not suited', array('%sortUsed' => $frequency_rank[$search_result->sortUsed]));
+    //   $message .= '<div class="messages status">' . $msg . '</div>';
+    // }
 
 
     if (isset($search_result) && $limit < $search_result->numTotalObjects) {
