diff --git a/modules/ting_search/plugins/content_types/search_result.inc b/modules/ting_search/plugins/content_types/search_result.inc
index b143d2af..feef3f18 100644
--- a/modules/ting_search/plugins/content_types/search_result.inc
+++ b/modules/ting_search/plugins/content_types/search_result.inc
@@ -91,10 +91,10 @@ function ting_search_search_result_content_type_render($subtype, $conf, $panel_a
       'rank_subject' => t('Subject')
     );
 
-    if (isset($search_result) && empty($conditions['sort']) && in_array($search_result->sortUsed, array_keys($frequency_rank))) {
-      $msg = t('Records are sorted by %sortUsed . Select another sort if it is not suited', array('%sortUsed' => $frequency_rank[$search_result->sortUsed]));
-      $message .= '<div class="messages status">' . $msg . '</div>';
-    }
+    // if (isset($search_result) && empty($conditions['sort']) && in_array($search_result->sortUsed, array_keys($frequency_rank))) {
+    //   $msg = t('Records are sorted by %sortUsed . Select another sort if it is not suited', array('%sortUsed' => $frequency_rank[$search_result->sortUsed]));
+    //   $message .= '<div class="messages status">' . $msg . '</div>';
+    // }
 
 
     if (isset($search_result) && $limit < $search_result->numTotalObjects) {
