diff --git a/profiles/ding2/ding2.info b/profiles/ding2/ding2.info
index 10e5bd4d..03684fab 100644
--- a/profiles/ding2/ding2.info
+++ b/profiles/ding2/ding2.info
@@ -33,12 +33,9 @@ dependencies[] = toolbar
 dependencies[] = ding_base
 dependencies[] = ding_user_frontend
 dependencies[] = ding_ting_frontend
-dependencies[] = ding_frontpage
 dependencies[] = ding_session_cache
 dependencies[] = ding_staff
 dependencies[] = bpi
-dependencies[] = ding_p2_installer

 ; Cache/performance modules
 dependencies[] = entitycache
diff --git a/profiles/ding2/ding2.install b/profiles/ding2/ding2.install
index b7f95916..81ae5f37 100644
--- a/profiles/ding2/ding2.install
+++ b/profiles/ding2/ding2.install
@@ -66,9 +66,6 @@ function ding2_install() {
   // Disable configurable timezones for users.
   variable_set('configurable_timezones', 0);
 
-  // Select the ding_frontpage as front page.
-  variable_set('site_frontpage', 'ding_frontpage');
-
   //
   // Enable ting search as default.
   //
diff --git a/profiles/ding2/ding2.profile b/profiles/ding2/ding2.profile
index 4f8e2d67..f8a9e529 100644
--- a/profiles/ding2/ding2.profile
+++ b/profiles/ding2/ding2.profile
@@ -230,7 +230,6 @@ function ding2_add_settings(&$install_state) {
     'ding_library',
     'ding_news',
     'ding_groups',
-    'ding_frontpage',
   );
 
   // Revert features.
diff --git a/profiles/ding2/modules/ding_frontend/ding_frontend.info b/profiles/ding2/modules/ding_frontend/ding_frontend.info
index de255767..772361bd 100644
--- a/profiles/ding2/modules/ding_frontend/ding_frontend.info
+++ b/profiles/ding2/modules/ding_frontend/ding_frontend.info
@@ -10,7 +10,6 @@ dependencies[] = conditional_styles
 dependencies[] = ctools
 dependencies[] = ding_page
 dependencies[] = ding_permissions
-dependencies[] = dynamic_background
 dependencies[] = entity
 dependencies[] = features
 dependencies[] = field_group
diff --git a/profiles/ding2/modules/ding_ting_frontend/ding_ting_frontend.info b/profiles/ding2/modules/ding_ting_frontend/ding_ting_frontend.info
index 2494dad6..59a54423 100644
--- a/profiles/ding2/modules/ding_ting_frontend/ding_ting_frontend.info
+++ b/profiles/ding2/modules/ding_ting_frontend/ding_ting_frontend.info
@@ -5,7 +5,6 @@ package = Ding!
 project = ding_ting_frontend
 dependencies[] = ctools
 dependencies[] = ding_availability
-dependencies[] = ding_ekurser
 dependencies[] = ding_entity
 dependencies[] = ding_facetbrowser
 dependencies[] = ding_frontend
@@ -23,7 +22,6 @@ dependencies[] = ting
 dependencies[] = ting_covers
 dependencies[] = ting_fulltext
 dependencies[] = ting_oembed_features
-dependencies[] = ting_proxy
 dependencies[] = ting_reference
 dependencies[] = ting_relation
 dependencies[] = ting_search
