diff --git a/profiles/ding2/modules/ting/ting.module b/profiles/ding2/modules/ting/ting.module
index 10ce0444..ed00f417 100644
--- a/profiles/ding2/modules/ting/ting.module
+++ b/profiles/ding2/modules/ting/ting.module
@@ -807,7 +807,13 @@ function ting_search_object_load($id) {
 function ting_single_object_render($entity_id, $view_mode = 'teaser') {
   // Load the ting objects found.
   $entity = entity_load('ting_object', array($entity_id));
-  $ting_object = ting_object_view(reset($entity), $view_mode);
+  $entity = reset($entity);
+
+  // Filter out missing records.
+  if (FALSE !== $entity && strpos($entity->getTitle(), 'Error: unknown/missing/inaccessible record:') === 0) {
+    return;
+  }
+  $ting_object = ting_object_view($entity, $view_mode);

   $build = array(
     'id' => $entity_id,
