diff --git a/profiles/ding2/modules/opensearch/includes/opensearch.search.inc b/profiles/ding2/modules/opensearch/includes/opensearch.search.inc
index b7d1af86..d68fdee9 100644
--- a/profiles/ding2/modules/opensearch/includes/opensearch.search.inc
+++ b/profiles/ding2/modules/opensearch/includes/opensearch.search.inc
@@ -683,8 +683,8 @@ function opensearch_search_autocomplete_suggestions(array $query) {
 
   $method = variable_get('ting_search_autocomplete_method', 'facets');
 
-  $rest_url = rtrim(variable_get('opensearch_search_autocomplete_suggestion_url', ''), '/').'/rest/';
-  $url = url($rest_url.$method, array('query' => $query));
+  $rest_url = rtrim(variable_get('opensearch_search_autocomplete_suggestion_url', ''), '/') . '/rest/';
+  $url = url($rest_url . $method, array('query' => $query));
   $headers = array('Accept' => 'application/json');
   $result = drupal_http_request($url, $headers);
 
