diff --git a/profiles/ding2/modules/opensearch/opensearch.client.inc b/profiles/ding2/modules/opensearch/opensearch.client.inc
index c6e49ebd..cee33dd4 100644
--- a/profiles/ding2/modules/opensearch/opensearch.client.inc
+++ b/profiles/ding2/modules/opensearch/opensearch.client.inc
@@ -403,7 +403,7 @@ function opensearch_execute(TingClientRequest $request) {
 
     // Update the cache with the raw response object from the data well for
     // faster processing. Also set the static cache.
-    _opensearch_cache_insert($request, $response, $opensearch_static_fast);
+    _opensearch_cache_insert($request, $response, $opensearch_static_fast, $cid);
 
     // To speed up the entity module, which will trigger an search for each
     // collection. Try and predict the search request an warm up the cache.
@@ -520,9 +520,11 @@ function _opensearch_predict_cache(array $collections, array &$opensearch_static
  *   The response to the request.
  * @param array $opensearch_static_fast
  *   The fast static cache used by opensearch_execute().
+ * @param string $cid
+ *   Cache id.
  */
-function _opensearch_cache_insert(TingClientRequest $request, $response, array &$opensearch_static_fast) {
-  $cid = _opensearch_cache_id($request);
+function _opensearch_cache_insert(TingClientRequest $request, $response, array &$opensearch_static_fast, $cid = '') {
+  $cid = empty($cid) ? _opensearch_cache_id($request) : $cid;
 
   // Update the cache with the raw response object from the data well for
   // faster processing. Also set the static cache.
