diff --git a/modules/opensearch/includes/opensearch.search.inc b/modules/opensearch/includes/opensearch.search.inc
index db3f5ee1f..7486e4b09 100644
--- a/modules/opensearch/includes/opensearch.search.inc
+++ b/modules/opensearch/includes/opensearch.search.inc
@@ -703,7 +703,10 @@ function opensearch_search_autocomplete_suggestions(array $query) {
     return $items;
   }
 
-  $url = url($rest_url, array('query' => $query));
+  // Make stuff work with http://opensuggestion.addi.dk/b3.5_2.0/#facets
+  $rest_url = rtrim($rest_url, '/') . '/rest/';
+  $query['query'] = $query['q'];
+  $url = url($rest_url . 'facets', array('query' => $query));
   $headers = array('Accept' => 'application/json');
   $result = drupal_http_request($url, $headers);
 
@@ -728,6 +731,9 @@ function opensearch_search_autocomplete_suggestions(array $query) {
         $items[] = $suggestion['all'][0];
       }
     }
+    elseif (!empty($result_data['suggestions'])) {
+      $items = array_column($result_data['suggestions'], 'suggestion');
+    }
   }
 
   return $items;
