diff --git a/profiles/ding2/modules/ding_base/ding_base.info b/profiles/ding2/modules/ding_base/ding_base.info
index 30f0fbb2..ffa5a140 100644
--- a/profiles/ding2/modules/ding_base/ding_base.info
+++ b/profiles/ding2/modules/ding_base/ding_base.info
@@ -2,7 +2,6 @@ name = Ding base
 core = 7.x
 package = Ding!
 project = ding_base
-dependencies[] = autologout
 dependencies[] = ctools
 dependencies[] = ding_permissions
 dependencies[] = jquery_update
@@ -20,34 +19,6 @@ configure = admin/config/ding
 features[ctools][] = page_manager:pages_default:1
 features[ctools][] = strongarm:strongarm:1
 features[features_api][] = api:2
-features[variable][] = autologout_enforce_admin
-features[variable][] = autologout_inactivity_message
-features[variable][] = autologout_max_timeout
-features[variable][] = autologout_message
-features[variable][] = autologout_no_dialog
-features[variable][] = autologout_padding
-features[variable][] = autologout_redirect_url
-features[variable][] = autologout_role_1
-features[variable][] = autologout_role_1_timeout
-features[variable][] = autologout_role_2
-features[variable][] = autologout_role_2_timeout
-features[variable][] = autologout_role_3
-features[variable][] = autologout_role_3_timeout
-features[variable][] = autologout_role_4
-features[variable][] = autologout_role_4_timeout
-features[variable][] = autologout_role_5
-features[variable][] = autologout_role_5_timeout
-features[variable][] = autologout_role_6
-features[variable][] = autologout_role_6_timeout
-features[variable][] = autologout_role_7
-features[variable][] = autologout_role_7_timeout
-features[variable][] = autologout_role_8
-features[variable][] = autologout_role_8_timeout
-features[variable][] = autologout_role_9
-features[variable][] = autologout_role_9_timeout
-features[variable][] = autologout_role_logout
-features[variable][] = autologout_timeout
-features[variable][] = autologout_use_watchdog
 features[variable][] = bpi_content_type
 features[variable][] = bpi_field_body
 features[variable][] = bpi_field_materials
diff --git a/profiles/ding2/modules/ding_base/ding_base.strongarm.inc b/profiles/ding2/modules/ding_base/ding_base.strongarm.inc
index 197629d4..ba91d767 100644
--- a/profiles/ding2/modules/ding_base/ding_base.strongarm.inc
+++ b/profiles/ding2/modules/ding_base/ding_base.strongarm.inc
@@ -24,206 +24,6 @@ function ding_base_strongarm() {
   );
   $export['admin_menu_components'] = $strongarm;
 
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_enforce_admin';
-  $strongarm->value = 1;
-  $export['autologout_enforce_admin'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_inactivity_message';
-  $strongarm->value = 'Af sikkerhedsmæssige årsager er du blevet logget ud, da du har været inaktiv de seneste 15 minutter
-
-';
-  $export['autologout_inactivity_message'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_max_timeout';
-  $strongarm->value = '172800';
-  $export['autologout_max_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_message';
-  $strongarm->value = 'Af sikkerhedsmæssige årsager vil du automatisk blive logget ud om et øjeblik. Vil du fortsætte med at være logget ind?
-
-';
-  $export['autologout_message'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_no_dialog';
-  $strongarm->value = 0;
-  $export['autologout_no_dialog'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_padding';
-  $strongarm->value = '15';
-  $export['autologout_padding'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_redirect_url';
-  $strongarm->value = '<front>';
-  $export['autologout_redirect_url'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_1';
-  $strongarm->value = 0;
-  $export['autologout_role_1'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_1_timeout';
-  $strongarm->value = '1800';
-  $export['autologout_role_1_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_2';
-  $strongarm->value = 0;
-  $export['autologout_role_2'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_2_timeout';
-  $strongarm->value = '1800';
-  $export['autologout_role_2_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_3';
-  $strongarm->value = 0;
-  $export['autologout_role_3'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_3_timeout';
-  $strongarm->value = '0';
-  $export['autologout_role_3_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_4';
-  $strongarm->value = 0;
-  $export['autologout_role_4'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_4_timeout';
-  $strongarm->value = '1800';
-  $export['autologout_role_4_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_5';
-  $strongarm->value = 0;
-  $export['autologout_role_5'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_5_timeout';
-  $strongarm->value = '1800';
-  $export['autologout_role_5_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_6';
-  $strongarm->value = 0;
-  $export['autologout_role_6'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_6_timeout';
-  $strongarm->value = '1800';
-  $export['autologout_role_6_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_7';
-  $strongarm->value = 0;
-  $export['autologout_role_7'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_7_timeout';
-  $strongarm->value = '1800';
-  $export['autologout_role_7_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_8';
-  $strongarm->value = 1;
-  $export['autologout_role_8'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_8_timeout';
-  $strongarm->value = '900';
-  $export['autologout_role_8_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_9';
-  $strongarm->value = 1;
-  $export['autologout_role_9'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_9_timeout';
-  $strongarm->value = '1800';
-  $export['autologout_role_9_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_role_logout';
-  $strongarm->value = 1;
-  $export['autologout_role_logout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_timeout';
-  $strongarm->value = '1800';
-  $export['autologout_timeout'] = $strongarm;
-
-  $strongarm = new stdClass();
-  $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
-  $strongarm->api_version = 1;
-  $strongarm->name = 'autologout_use_watchdog';
-  $strongarm->value = 0;
-  $export['autologout_use_watchdog'] = $strongarm;
-
   $strongarm = new stdClass();
   $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
   $strongarm->api_version = 1;
diff --git a/sites/all/modules/reol_base/reol_base.install b/sites/all/modules/reol_base/reol_base.install
index d7c04d47..0497524b 100644
--- a/sites/all/modules/reol_base/reol_base.install
+++ b/sites/all/modules/reol_base/reol_base.install
@@ -514,3 +514,13 @@ function reol_base_update_7148() {
 function reol_base_update_7149() {
   features_revert_module('ding_ting_frontend');
 }
+
+/**
+ * Disable autologout module.
+ */
+function reol_base_update_7150() {
+  // It don't makes sens to auto logout users will listening or reading an online book.
+  if (module_exists('autologout')) {
+    module_disable(array('autologout'), FALSE);
+  }
+}
