diff --git a/modules/ting_search_carousel/ting_search_carousel.module b/modules/ting_search_carousel/ting_search_carousel.module
index 4a943ff..2951e0a 100644
--- a/modules/ting_search_carousel/ting_search_carousel.module
+++ b/modules/ting_search_carousel/ting_search_carousel.module
@@ -106,6 +106,45 @@ function ting_search_carousel_block_view($delta) {
 }
 
 /**
+ * Implements hook_cron().
+ *
+ * Touch all existing cover files, to stop ting_cover_cron() from deleting them,
+ * or fetch missing covers (they must exists as they wouldn't have been cache
+ * otherwise).
+ *
+ * An alternative would be to simply clear our own cache at the same time. Has
+ * the advantage that the data doesn't grow stale for too long, in the case of
+ * updates to the datawell, but means hitting the well harder.
+ */
+function ting_search_carousel_cron() {
+  if ($cache = cache_get('ting_search_carousel_result')) {
+    $data = $cache->data;
+
+    $ids = array();
+    foreach ($data as $index) {
+      foreach ($index as $item) {
+        $path = ting_covers_object_path($item->id);
+        if (file_exists($path)) {
+          // Touch the file to keep ting_cover_cron() from deleting it. The
+          // Drupal stream wrapper doesn't support touch, so get the real path.
+          touch(drupal_realpath($path));
+        }
+        else {
+          // Fetch the cover.
+          $ids[] = $item->id;
+        }
+      }
+    }
+
+    if ($ids) {
+      $entities = ting_object_load_multiple($ids);
+      ting_covers_load($entities);
+    }
+  }
+}
+
+
+/**
  * Implements hook_ctools_plugin_directory().
  */
 function ting_search_carousel_ctools_plugin_directory($module, $plugin) {
@@ -145,13 +184,15 @@ function ting_search_carousel_do_request($index) {
   }
 
   // Loop over the search results collecting basic object information.
-  $local_ids = array();
+  $ids = array();
   $objects = array();
   foreach ($collections as $collection) {
     foreach ($collection->collections as $ting_collection) {
       foreach ($ting_collection->reply->objects as $object) {
-        $local_ids[] = $object->localId;
-        $objects[$object->localId] = array(
+        $ids[$object->id] = array(
+          'image_style' => 'ting_search_carousel',
+        );
+        $objects[$object->id] = array(
           'id' => $object->id,
           'title' => isset($object->record['dc:title'][''][0]) ? $object->record['dc:title'][''][0] : '',
           'creator' => isset($object->record['dc:creator']['oss:aut'][0]) ? $object->record['dc:creator']['oss:aut'][0] : '',
@@ -160,49 +201,20 @@ function ting_search_carousel_do_request($index) {
     }
   }
 
-  // Load helper function form the covers module.
-  module_load_include('pages.inc', 'ting_covers');
-
-  // Get cover service client.
-  $service = new AdditionalInformationService(
-          variable_get('addi_wsdl_url'),
-          variable_get('addi_username'),
-          variable_get('addi_group'),
-          variable_get('addi_password'));
-
   // Initialize items array.
   $items = array();
 
   // Get covers for the ids.
-  $covers = $service->getByFaustNumber($local_ids);
+  $covers = ting_covers_load($ids);
 
   // Loop over the fetched covers and build items.
-  foreach ($covers as $local_id => $cover) {
-    // Build cover filename.
-    $filename = 'public://ting_search_carousel/' . md5($local_id) . '.jpg';
-
-    // Check if file have been downloaded.
-    if (!file_exists($filename)) {
-      // Extract the image source url.
-      $source_url = FALSE;
-      if ($cover->detailUrl) {
-        $source_url = $cover->detailUrl;
-      }
-      elseif ($cover->thumbnailUrl) {
-        $source_url = $cover->thumbnailUrl;
-      }
-
-      // Download the cover image and get local uri.
-      $file = _ting_covers_pages_fetch_image($filename, $source_url);
-      $filename = isset($file->uri) ? $file->uri : $filename;
-    }
-
+  foreach ($covers as $id => $cover_info) {
     // Build object with the information collected.
     $item = new stdClass();
-    $item->id = $objects[$local_id]['id'];
-    $item->title = $objects[$local_id]['title'];
-    $item->creator = $objects[$local_id]['creator'];
-    $item->image = image_style_url('ting_search_carousel', $filename);
+    $item->id = $objects[$id]['id'];
+    $item->title = $objects[$id]['title'];
+    $item->creator = $objects[$id]['creator'];
+    $item->image = $cover_info['url'];
 
     $items[] = $item;
   }
