diff --git a/profiles/ding2/modules/ting_material_details/ting_material_details.field.inc b/profiles/ding2/modules/ting_material_details/ting_material_details.field.inc
index 5a2238d8..08f0b31e 100644
--- a/profiles/ding2/modules/ting_material_details/ting_material_details.field.inc
+++ b/profiles/ding2/modules/ting_material_details/ting_material_details.field.inc
@@ -617,10 +617,10 @@ function ting_material_details_field_formatter_view($entity_type, $entity, $fiel
           }
           $links = array();
           foreach ($val as $value) {
-            $links[] = l($value, 'search/ting/phrase.subject=' . $value);
+            $links[] = l($value, 'search/ting/phrase.subject=' . $value, array('attributes' => array('class' => array('subject'))));
           }
           $element[$delta] = array(
-            '#markup' => implode(', ', $links),
+            '#markup' => implode(' ', $links),
           );
           break;
 
