diff --git a/profiles/ding2/modules/ting_search_carousel/ting_search_carousel.module b/profiles/ding2/modules/ting_search_carousel/ting_search_carousel.module
index 29631430..18361c46 100644
--- a/profiles/ding2/modules/ting_search_carousel/ting_search_carousel.module
+++ b/profiles/ding2/modules/ting_search_carousel/ting_search_carousel.module
@@ -286,7 +286,7 @@ function _ting_search_carousel_find_entities_with_covers($query, $start, $size,
   // Search offset must be divisible by chunk size.
   // We cannot map a situation where offset is not divisible by chunk size to
   // the search abstraction layer paging model.
-  if (!$start % $size === 0) {
+  if ($start % $size !== 0) {
     throw new InvalidArgumentException('Offset to start from % search chunk size should not have remainders left.');
   }
 
@@ -323,7 +323,7 @@ function _ting_search_carousel_find_entities_with_covers($query, $start, $size,
     $entities_found_with_covers[$id] = $entities[$id];
   }
 
-  return array($entities_found_with_covers, $start, $finished);
+  return array($entities_found_with_covers, $start + $size, $finished);
 }
 
 /**
