diff --git a/ding2.make b/ding2.make
index b7ee052..19e87ca 100644
--- a/ding2.make
+++ b/ding2.make
@@ -156,24 +156,16 @@ projects[i18n][version] = "1.11"
 projects[maintenance_mode_api][subdir] = "contrib"
 projects[maintenance_mode_api][version] = "1.0-beta1"
 
-; This version of media is tested to work with both images and videos.
 projects[media][type] = "module"
 projects[media][subdir] = "contrib"
-projects[media][download][type] = "git"
-projects[media][download][url] = "http://git.drupal.org/project/media.git"
-projects[media][download][revision] = "c3cda2b"
-; Fixed issue where "insert" fails, see https://www.drupal.org/node/2184475.
-projects[media][patch][] = "https://www.drupal.org/files/issues/media_popup_trigger_some_js-2184475-6.patch"
+projects[media][version] = "2.0-alpha4"
 
 projects[media_vimeo][subdir] = "contrib"
 projects[media_vimeo][version] = "2.0-rc1"
 
 projects[media_youtube][type] = "module"
 projects[media_youtube][subdir] = "contrib"
-projects[media_youtube][download][type] = "git"
-projects[media_youtube][download][url] = "http://git.drupal.org/project/media_youtube.git"
-projects[media_youtube][download][revision] = "ca46aba"
-projects[media_youtube][patch][] = "http://drupal.org/files/issues/provide-access-wrapper-1823376-6.patch"
+projects[media_youtube][version] = "3.0"
 
 projects[memcache][subdir] = "contrib"
 projects[memcache][version] = "1.0"
