diff --git a/profiles/ding2/modules/ding_user_frontend/ding_user_frontend.module b/profiles/ding2/modules/ding_user_frontend/ding_user_frontend.module
index d6f260ac..4356683e 100644
--- a/profiles/ding2/modules/ding_user_frontend/ding_user_frontend.module
+++ b/profiles/ding2/modules/ding_user_frontend/ding_user_frontend.module
@@ -242,22 +242,8 @@ function ding_user_frontend_build_menu($path = NULL) {
       }
     }
 
-    // We need this menu-item in order for the menu to appear on the page.
-    // We unset this menu-item because the menu header links to the same
-    // page, and we don't want it to also be a link in the menu.
-    unset($menu['user/%/view']);
-
     $menu['user/%/edit']['#weight'] = 7;
 
-    // Add logout to menu.
-    $logout_item = array();
-    $logout_item = $menu_item_defaults;
-    $logout_item['#title'] = t('Log out');
-    $logout_item['#href'] = 'user/logout';
-    $logout_item['#weight'] = 9;
-    $logout_item['#attributes']['class'] = 'user-log-out';
-    $menu[$logout_item['#href']] = $logout_item;
-
     drupal_alter('ding_user_frontend_build_menu', $menu, $menu_item_defaults);
   }
 
